deploy/test-environments/fleet_api/src/api/agent_policy_api.py

```diff
@@ -40,6 +41,37 @@ def create_agent_policy(cfg: Munch, json_policy: dict) -> str:
         raise api_ex
 
 
+def update_agent_policy(cfg: Munch, policy_id, json_policy: dict):
+    """This function updates an agent policy
+
+    Args:
+        cfg (Munch): Config object containing authentication data.
+        policy_id (str): Policy id to be updated.
+        json_policy (dict): Data for the agent policy to be updated.
+
+    Raises:
+        APICallException: If the API call fails or returns a non-200 status code.
+    """
+    # pylint: disable=duplicate-code
+    url = f"{cfg.kibana_url}/api/fleet/agent_policies/{policy_id}"
+
+    try:
+        perform_api_call(
+            method="PUT",
+            url=url,
+            auth=cfg.auth,
+            params={"json": json_policy},
+        )
+        logger.info(
+            f"Agent policy '{policy_id}' for integration '{json_policy.get('name', '')}' has been updated",
+        )
+    except APICallException as api_ex:
```

Comment:
@amirbenun in our case, removing `try-except` blocks wouldn't make things clearer. We raise `custom-api-call` exceptions that give us the `status code` and a `message` when a response other than 200 is returned.

The main difference is that without the `try-except` blocks, the script would abruptly stop if a call doesn't return a 200 status code, instead of just logging the error and continuing the script run.